Sources/Charcoal/Filters/List/ListFilterViewController.swift

```diff
@@ -63,12 +86,57 @@ public final class ListFilterViewController: FilterViewController {
     private func setup() {
         view.insertSubview(tableView, belowSubview: bottomButton)
 
-        NSLayoutConstraint.activate([
+        let sharedTableViewConstraints = [
             tableView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
-            tableView.topAnchor.constraint(equalTo: view.topAnchor),
             tableView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
             tableView.bottomAnchor.constraint(equalTo: bottomButton.topAnchor),
-        ])
+        ]
+
+        if filter.subfilters.count >= searchbarSubfilterThreshold {
+            view.addSubview(searchBar)
+
+            topShadowViewBottomAnchor.isActive = false
+
+            let searchBarConstraints = [
+                searchBar.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: .spacingS),
+                searchBar.topAnchor.constraint(equalTo: view.topAnchor),
+                searchBar.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -.spacingS),
+
+                topShadowView.bottomAnchor.constraint(equalTo: searchBar.bottomAnchor),
+                tableView.topAnchor.constraint(equalTo: searchBar.bottomAnchor),
+            ]
+
+            NSLayoutConstraint.activate(sharedTableViewConstraints + searchBarConstraints)
+        } else {
+            NSLayoutConstraint.activate(sharedTableViewConstraints + [tableView.topAnchor.constraint(equalTo: view.topAnchor)])
+        }
+    }
+
+    // MARK: - Overrides
+
+    public func scrollViewWillBeginDragging(_ scrollView: UIScrollView) {
+        super.scrollViewDidScroll(scrollView)
+        searchBar.endEditing(true)
+    }
+
+    // MARK: - Actions
+
+    @objc private func adjustForKeyboard(notification: Notification) {
+        guard let keyboardValue = notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue else { return }
+
+        var keyboardHeight = view.convert(keyboardValue.cgRectValue, from: view.window).height
+
+        if #available(iOS 11.0, *) {
```

Comment:
uh, do we support less than iOS 11 in charcoal? I guess we shouldn't as we don't test it

Reply:
No, I think you're right. Copied this code from another place, so I'll change there as well